pixlie_ai/src/entity/web/scraper.rs

```diff
@@ -47,173 +49,46 @@ impl<'a> Traverser<'a> {
-                                        )?;
-                                    }
-                                }
-                                _ => {}
-                            }
-                        }
-                    }
-                }
-                "img" => {
-                    if let Some(src) = element.value().attr("src") {
-                        if src.starts_with("http") {
-                            self.add_text_node(
-                                vec![NodeLabel::Image, NodeLabel::Metadata],
-                                src,
-                                false,
-                            )?;
-                        } else {
-                            let base_url = self.webpage_url.join("/")?;
-                            let src = base_url.join(src)?;
-                            self.add_text_node(
-                                vec![NodeLabel::Image, NodeLabel::Metadata],
-                                &src.to_string(),
-                                false,
-                            )?;
+                        if let Some(name) = element.value().attr("name") {
+                            self.update_metadata_node(&name, content)?;
                         }
                     }
                 }
                 "link" => {
                     if let Some(href) = element.value().attr("href") {
                         if let Some(rel) = element.value().attr("rel") {
                             if rel.contains("icon") {
-                                if href.starts_with("http") {
-                                    self.add_text_node(
-                                        vec![NodeLabel::Logo, NodeLabel::Metadata],
-                                        href,
-                                        false,
-                                    )?;
-                                } else {
-                                    let base_url = self.webpage_url.join("/")?;
-                                    let href = base_url.join(href)?;
-                                    self.add_text_node(
-                                        vec![NodeLabel::Logo, NodeLabel::Metadata],
-                                        &href.to_string(),
-                                        false,
-                                    )?;
-                                }
+                                self.update_metadata_node("image", href)?;
                             }
                         }
                     }
                 }
-                "title" => self.add_text_node(
-                    vec![NodeLabel::Title, NodeLabel::Metadata],
-                    &element.text().collect::<Vec<&str>>().join(""),
-                    false,
-                )?,
+                "img" => {
+                    if let Some(src) = element.value().attr("src") {
+                        self.update_metadata_node("image", src)?;
```

Comment:
The number of times this function is getting called in same part of the code is potentially ~10 writes to the same node, in cases of abundant metadata.
It would be a good idea to gather all the data and write it at once.
Moreover, IIRC we were discussing storing this metadata as a property of the webpage node. Though, I feel it's better to keep it this way - in a separate node.